src/store/ChartStore.js

```diff
@@ -231,25 +231,67 @@ class ChartStore {
-                this.restoreLayout(stxx, layoutData);
-
-                this.setActiveSymbols(active_symbols);
-
-                if (initialSymbol && !(layoutData && layoutData.symbols)) {
-                    this.changeSymbol(initialSymbol);
-                } else if (stxx.chart.symbol) {
-                    this.currentActiveSymbol = stxx.chart.symbolObject;
-                    stxx.chart.yAxis.decimalPlaces = stxx.chart.symbolObject.decimal_places;
-                    this.categorizedSymbols = this.categorizeActiveSymbols();
-                    if (onSymbolChange) { onSymbolChange(this.currentActiveSymbol); }
-                } else {
-                    this.changeSymbol(this.defaultSymbol);
-                }
-
+                const today_utc = Date.UTC(
+                        (new Date()).getUTCFullYear(),
+                        (new Date()).getUTCMonth(),
+                        (new Date()).getUTCDate(),
+                        (new Date()).getHours(),
+                        (new Date()).getMinutes(),
+                        (new Date()).getSeconds(),
+                    ),
+                    timeToEpochGMT = (time_stirng) => {
+                        let today = new Date(),
+                            time = time_stirng.split(':');
+                        return Date.UTC(today.getUTCFullYear(), today.getUTCMonth(), today.getUTCDate(), time[0], time[1], time[2]);
+                    },
+                    updateActiveSymbols = () => {
+                        api.getTradingTimes().then((data) => {
+                            data.trading_times.markets.forEach((market) => {
+                                market.submarkets.forEach((submarket) => {
+                                    submarket.symbols.forEach((symbol) => {
+                                        const foundSymbol = active_symbols.find(item => item.symbol === symbol.symbol);
+                                        if (foundSymbol) {
+                                            let isOpen = false;
+                                            for (let i = 0; i <= symbol.times.open.length; i++) {
+                                                if (
+                                                    symbol.times.open[i] &&
+                                                symbol.times.close[i] &&
+                                                today_utc >= timeToEpochGMT(symbol.times.open[i]) &&
+                                                today_utc <= timeToEpochGMT(symbol.times.close[i])
+                                                ) {
+                                                    isOpen = true;
+                                                }
+                                            }
+                                            foundSymbol.is_open = isOpen;
+                                        }
+                                    });
+                                });
+                            });
```

Comment:
The symbols you have modified are not the ones used by the `CategoryDisplay`'s. You need to update `categorizedSymbols`:

```js
this.categorizedSymbols = this.categorizeActiveSymbols();
```

Also `categorizeActiveSymbols()` uses data from `activeSymbols`, so modifying `active_symbols` won't affect anything.